### PR TITLE
OPENNLP-1045: Git documentation for developers

### PR DESCRIPTION
**Pull request by kinow**
Add documentation for development with Git (at ASF, GitHub, etc) for OpenNLP.

Inspired mainly at the Cordova docs [1]. Did not include `--squash` option, which the Mahout docs mention [2] as I never used it :-)

Feel free to submit suggestions or update the pull request if you have access to.

Cheers
Bruno

[1] https://github.com/apache/cordova-coho/blob/master/docs/processing-pull-requests.md
[2] https://mahout.apache.org/developers/github.html

### CODE REVIEW

**Review by jzonthemtn — APPROVED** [review]

**Comment by kinow** [comment]
@smarthi I remember you had some Git docs from Mahout (I think) that you thought we should use too. Are you happy too with us using this version in this pull request for now, and then start working on top of that?

File: src/main/jbake/content/using-git.ad [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+the GitHub user interface, or add a comment to the JIRA ticket &mdash; if any.
+
+Once you are happy with the changes, you can check out the master branch, and merge the pull request. Remember
+to make sure the **branch has been rebase'd against master**, and also that all tests pass.
+
+    git checkout master
+    git merge --ff-only <BRANCHNAME>
+    git push origin master
+
+In case other commits happened after the pull request was submitted, you must ask the user to rebase the pull
```
Comment (kottmann) [inline src/main/jbake/content/using-git.ad]: The easiest thing is to rebase yourself, the contributor only needs to be involved if there are conflicts.

File: src/main/jbake/content/using-git.ad [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+request against the master branch, and squash his commit. An alternative for that, is to rebase and squash the commits
+yourself, when there is no feedback from the user.
+
+    git checkout <BRANCHNAME>
+    git rebase master
+    # squash and amend the commit message if necessary...
```
Comment (kottmann) [inline src/main/jbake/content/using-git.ad]: We only accept PRs that have a single commit, then we don't have to deal with that during merging.

File: src/main/jbake/content/using-git.ad (hunk 1 above)
Comment (kinow) [reply src/main/jbake/content/using-git.ad]: Good point. Let me re-write that quickly now (will squash commits too)

File: src/main/jbake/content/using-git.ad [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    git merge --ff-only <BRANCHNAME>
+    git push origin master
+
+### Stale Pull Requests
+
+You can automatically close a pull request when merging, by amending the commit message. Make sure to use the following
+syntax in your commit message.
+
+    Merging branch <BRANCHNAME>
+    OPENNLP-<JIRA-ID>: This closes #<PULL-ID>
```
Comment (kottmann) [inline src/main/jbake/content/using-git.ad]: here we usually write closes #xyz and that's it, preferable not in the subject line

Comment (kinow) [reply src/main/jbake/content/using-git.ad]: Oh, my mistake. The OPENNLP-<JIRA-ID> was supposed to be on the line above where it is now.

**Pull request closed without merging**